lib/puppetserver/ca/action/prune.rb

```diff
@@ -65,10 +66,12 @@ def run(inputs)
 
         def prune_CRLs(crl_list)
           number_of_removed_duplicates = 0
+          number_of_certificates = 0
 
           crl_list.each do |crl|
```

Comment:
So this is potentially looping through multiple CRLs, right? I think we might want to do the logging within this method, rather than in `run`, so that we can log separately for each CRL we prune - I'm not sure it's valuable to add together all the CRL entries from all CRLs.

Reply:
hmm though I guess this does mirror what we do for the duplicates.. Maybe it makes sense to adjust `prune_CRLs` to just prune a single CRL?

Reply:
I believe that `prune_CRLs` only operate on Puppet's CRL which there should be only one (please correct me if I am wrong here) so I opted to do it the way that I did.

Reply:
The variable `puppet_crl` should be Puppet's CRL and that should be a single CRL.

Reply:
The way we're calling it, yes, we only pass a list of one CRL, but this method currently allows for a list of CRLs, right? Since we don't need that functionality, we could just update `prune_CRLs` to be `prune_crl` singular and just take a singular CRL.

Reply:
I see what you meant, I think I had that thought in mind when I was coding before and since `select` return an array even if it's just one item hence why I went with a "safe" option of using a list of CRLs just in case that there might be more than one Puppet's CRL (even though that should not happen).

Reply:
yeah, we could add a check that the returned array from `select` only contains one element and then just use `first` to pass the single CRL.

Reply:
Since now we're just pruning one CRL, do you think it's a good idea to log the number of certificate in `prune_CRL`? I sort of reserved debug logging for `prune_CRL`.

Reply:
I think it's probably okay to keep it in `run` if we'll definitely only be operating on a single CRL.